### PR TITLE
Private Events  completion

### PR DESCRIPTION
**Pull request by RNtaate**
## What does this PR do?
- [X] Creates User, Event, and Event_attendance models.
- [X] Creates Associations amongst User, Event, and Event_attendance models
- [X] Creates users, events, and sessions controllers.
- [X] Creates simple authentication of users.
- [X] Creates view templates for users, events, and sessions controllers.
- [X] Creates integration tests for project features.

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<h1>Welcome <%= @user.name %> </h1>
+
+<h3>Previously attended events</h3>
+<% @user.attended_events.where("date < ?", Date.today).each do |event|%>
+  <%= link_to event.description, event_path(event.id) %><br>
+<% end %>
```
Comment (maelfosso) [inline app/views/users/show.html.erb]: The better to do it it's by using Event scope
```rb
@user.attended_events.past
```
Create a variable in your action controller and use it here (in that view)

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+<h3>Upcoming events</h3>
+<% @user.attended_events.where("date >= ?", Date.today).each do |event|%>
+  <%= link_to event.description, event_path(event.id) %><br>
+<% end %>
```
Comment (maelfosso) [inline app/views/users/show.html.erb]: The better to do it it's by using Event scope
```rb
@user.attended_events.upcoming
```
Create a variable in your action controller and use it here (in that view)
```rb
def show
  @upcoming = @user.attended_events.upcoming
  ...
  ...
end
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for the job done so far.
Your application is working like a charm.
However, there are some missing requirements

1. You didn't write integrations tests, as requested [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png), for **events** and **users** management. Learn more about [here](https://hackernoon.com/how-to-build-awesome-integration-tests-with-capybara-j9333y68)

For the unit test, I suggest you use the **shoulda-matcher** gem. It's easy and fast.

2. It's only possible to attend an event during its creation. Please, add the possibility to attend the event after the event is created

3. As indicated [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png), the user show page must also display the event the user has created which is not the case in your application. Kindly fix that, please.

4. Read the comment left below

Kindly make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

**Comment by RNtaate** [comment]
## Changes made 
- [X] Added possibility to attend an event after the event is created.
- [X] Added created events to a user's show page.
- [X] Used Event scope to represent querries in view templates.
- [X] Made changes to integration tests as required by project specifications.

**Review by Dandush03 — APPROVED** [review]
# Status: :rocket: Project Approved :green_circle: 
### Great work, keep it up
## :tophat: Happy Coding :computer:

**Pull request merged**